tests/src/test/spark350/scala/com/nvidia/spark/rapids/iceberg/datagen.scala

```diff
@@ -0,0 +1,406 @@
+  }
+
+  private final def fillPool(): Array[A] = {
+    val ret = new Array[A](poolSize)
+    for (i <- 0 until poolSize) {
+      val isNull = random.nextBoolean()
+      if (isNull) {
+        ret(i) = null.asInstanceOf[A]
+      } else {
+        ret(i) = genNext(random)
+      }
+    }
+    ret
+  }
+
+  override def toHostColumnVector(size: Int): HostColumnVector = {
+    val data = generate(size)
+    toHostColumnVector(data)
+  }
+}
+
+class FilePathDataGen(var curFilePath: String) extends ColumnDataGen[String] {
+  require(curFilePath != null, "File path cannot be null")
+
+  def setCurrentFilePath(filePath: String): Unit = {
+    require(filePath != null, "File path cannot be null")
+    curFilePath = filePath
+  }
+
+  def generate(size: Int): Array[String] = {
+    (0 until size)
+      .map(_ => curFilePath)
+      .toArray[String]
+  }
+
+  def toHostColumnVector(size: Int): HostColumnVector = {
+    val data = generate(size)
+    HostColumnVector.fromStrings(data: _*)
+  }
+
+  def poolValue(size: Int, addNull: Boolean = true): Array[String] = {
+    throw new UnsupportedOperationException("Pooled value not supported by FilePathDataGen")
+  }
+
+
+  def toPoolHostColumnVector(size: Int, addNull: Boolean = true): HostColumnVector = {
+    throw new UnsupportedOperationException("Pooled vector not supported by FilePathDataGen")
+  }
+}
+
+class RowIndexDataGen(var rowIdx: Int = 0) extends ColumnDataGen[JLong] {
+  def reset(): Unit = {
+    rowIdx = 0
+  }
+
+  override def generate(size: Int): Array[JLong] = {
+    val data = (rowIdx until rowIdx + size)
+      .map(r => JLong.valueOf(r))
+      .toArray
+    rowIdx += size
+    data
+  }
+
+  override def toHostColumnVector(size: Int): HostColumnVector = {
+    val data = generate(size)
+    HostColumnVector.fromBoxedLongs(data: _*)
+  }
+
+  override def poolValue(size: Int, addNull: Boolean): Array[JLong] = {
+    throw new UnsupportedOperationException("Pooled value not supported by RowIndexDataGen")
+  }
+
+  override def toPoolHostColumnVector(size: Int, addNull: Boolean): HostColumnVector = {
+    throw new UnsupportedOperationException("Pooled vector not supported by RowIndexDataGen")
+  }
+}
+
+class ConstantDataGen[A <: AnyRef : ClassTag](val value: A, toHostVec: Array[A] => HostColumnVector)
+  extends ColumnDataGen[A] {
+  override def generate(size: Int): Array[A] = {
+    Array.fill(size)(value)
+  }
+
+  override def toHostColumnVector(size: Int): HostColumnVector = {
+    val data = generate(size)
+    toHostVec(data)
+  }
+
+  override def poolValue(size: Int, addNull: Boolean): Array[A] = {
+    if (addNull) {
+      generate(size) :+ null.asInstanceOf[A]
+    } else {
+      generate(size)
+    }
+  }
+
+  override def toPoolHostColumnVector(size: Int, addNull: Boolean): HostColumnVector = {
+    toHostVec(poolValue(size, addNull))
+  }
+}
+
+object ColumnDataGen {
+  private val SEED = 1742356520L
+  val RANDOM = new Random(SEED)
```

Comment:
this is using a global random, if the cases running order is non-deterministic, then we can
not reproduce the cases when encountering an issue. It's better to use a new random for each data gen.